xml/System.Runtime.CompilerServices/Unsafe.xml

```diff
@@ -1339,10 +1601,69 @@ This check is conceptually similar to `(void*)(&left) < (void*)(&right)`.
         <Parameter Name="value" Type="T" RefType="out" Index="0" FrameworkAlternate="net-5.0;net-6.0;net-7.0" />
       </Parameters>
       <Docs>
-        <typeparam name="T">The type of the uninitialized object.</typeparam>
-        <param name="value">The uninitialized object.</param>
-        <summary>Bypasses definite assignment rules for a given value.</summary>
-        <remarks>To be added.</remarks>
+        <typeparam name="T">The type of the reference.</typeparam>
+        <param name="value">The reference whose initialization should be skipped.</param>
+        <summary>Bypasses definite assignment rules for a given reference.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This is typically used to avoid double-assignment when initializing union-type structs. Consider the following example, which produces a C# compiler error.
+
+```cs
+using System;
+using System.Runtime.InteropServices;
+
+public static MyUnionStruct MyMethod() {
+    MyUnionStruct value;
+    value.SomeIntField = 42;
+    return value; // CS0165: Use of unassigned local variable 'value'
+}
+  
+[StructLayout(LayoutKind.Explicit)]
+public struct MyUnionStruct
+{
+    [FieldOffset(0)]
+    public int SomeIntField;
+    
+    [FieldOffset(0)]
+    public byte SomeByteField;
+}
+```
+
+This error occurs because the compiler expects _all_ fields of the struct to be initialized before the struct is used or returned to the caller.
+
+One way to avoid this compiler error is to ensure the entire struct is zero-inited before individual fields are set, as shown in the below example.
```

Comment:
```suggestion
One way to avoid this compiler error is to ensure the entire struct is zero-initialized before individual fields are set, as shown in the following example.
```